learn-arai/app/hooks/useAuth.ts

```diff
@@ -0,0 +1,78 @@
+'use client';
+
+// this react hook responsible for checking if user is logged in.
+import { redirect } from 'next/navigation';
+
+import { useEffect, useState } from 'react';
+
+import { User } from './useUser';
+import { useUser } from './useUser';
+
+export const useAuth = () => {
+    const { addUser, removeUser, user } = useUser();
+    const [isAuthenticated, setIsAuthenticated] = useState<boolean>();
+    const [isSessionExpire, setIsSessionExpire] = useState<boolean>();
+
+    useEffect(() => {
+        checkSession();
+        if (isSessionExpire) {
+            setIsAuthenticated(true);
+        } else {
+            setIsAuthenticated(false);
+        }
+    }, [isSessionExpire]);
```

Comment:
isSessionExpire will update if checkSession is ran but checkSession is only run when isSessionExpire is updated?

Reply:
`checkSession` is await function, and useEffect is not allowed await. So every line will be executed simultaneously.
And isSession always gonna be null.

so I fixed by when `checkSession` is finished exercuted, and the value if update. then set isAuthenticated to `true`.

I seeee, I should I move `setIsAuthenticated` into `checkSession()`